syntax = "proto3";

package playground;

service Playground {
  rpc Ping(PingRequest) returns (PingReply) {}
  rpc SetIamPolicy(SetIamPolicyRequest) returns (SetIamPolicyReply) {}
  rpc GetIamPolicy(GetIamPolicyRequest) returns (GetIamPolicyReply) {}
  rpc CheckIamPolicy(CheckIamPolicyRequest) returns (CheckIamPolicyReply) {}
  rpc AddMember(AddMemberRequest) returns (AddMemberReply) {}
  rpc CreateMember(CreateMemberRequest) returns (CreateMemberReply) {}
  rpc DeleteMember(DeleteMemberRequest) returns (DeleteMemberReply) {}
  rpc AddResource(AddResourceRequest) returns (AddResourceReply) {}
  rpc DeleteResource(DeleteResourceRequest) returns (DeleteResourceReply) {}
}

message PingRequest {
  string data = 1;
}

message PingReply {
  string data = 1;
}

message AddMemberRequest {
}

message AddMemberReply {
}

message CreateMemberRequest {
}

message CreateMemberReply {
}

message DeleteMemberRequest {
}

message DeleteMemberReply {
}

message AddResourceRequest {
}

message AddResourceReply {
}

message DeleteResourceRequest {
}

message DeleteResourceReply{
}

message CheckIamPolicyRequest {
}

message CheckIamPolicyReply {

}

message SetIamPolicyRequest {

}

message SetIamPolicyReply {

}

message GetIamPolicyRequest {

}

message GetIamPolicyReply {

}
